lib/api/core/validation/index.js

```diff
@@ -1,690 +1,708 @@
-  }
+    if (fieldSpec.hasOwnProperty('multivalued')) {
+      if (!checkAllowedProperties(fieldSpec.multivalued, ['value', 'minCount', 'maxCount'])) {
+        throwOrStoreError(`In ${indexName}.${collectionName}.${fieldName}, the multivalued field specification has invalid properties.`, verboseErrors, errors);
+      }
 
-  if (fieldSpec.hasOwnProperty('multivalued')) {
-    if (!checkAllowedProperties(fieldSpec.multivalued, ['value', 'minCount', 'maxCount'])) {
-      throwOrStoreError(`In ${indexName}.${collectionName}.${fieldName}, the multivalued field specification has invalid properties.`, verboseErrors, errors);
-    }
+      if (!fieldSpec.multivalued.hasOwnProperty('value')) {
+        throwOrStoreError(`In ${indexName}.${collectionName}.${fieldName}, "value" is a mandatory property for multivalued field specification.`, verboseErrors, errors);
+      }
 
-    if (!fieldSpec.multivalued.hasOwnProperty('value')) {
-      throwOrStoreError(`In ${indexName}.${collectionName}.${fieldName}, "value" is a mandatory property for multivalued field specification.`, verboseErrors, errors);
-    }
+      if (!fieldSpec.multivalued.value && (fieldSpec.multivalued.hasOwnProperty('minCount'))) {
+        throwOrStoreError(`In ${indexName}.${collectionName}.${fieldName}, "minCount" is not valid when multivalued field is disabled.`, verboseErrors, errors);
+      }
 
-    if (!fieldSpec.multivalued.value && (fieldSpec.multivalued.hasOwnProperty('minCount'))) {
-      throwOrStoreError(`In ${indexName}.${collectionName}.${fieldName}, "minCount" is not valid when multivalued field is disabled.`, verboseErrors, errors);
-    }
+      if (!fieldSpec.multivalued.value && (fieldSpec.multivalued.hasOwnProperty('maxCount'))) {
+        throwOrStoreError(`In ${indexName}.${collectionName}.${fieldName}, "maxCount" is not valid when multivalued field is disabled.`, verboseErrors, errors);
+      }
 
-    if (!fieldSpec.multivalued.value && (fieldSpec.multivalued.hasOwnProperty('maxCount'))) {
-      throwOrStoreError(`In ${indexName}.${collectionName}.${fieldName}, "maxCount" is not valid when multivalued field is disabled.`, verboseErrors, errors);
+      if (fieldSpec.multivalued.hasOwnProperty('minCount') && fieldSpec.multivalued.hasOwnProperty('maxCount') && fieldSpec.multivalued.minCount > fieldSpec.multivalued.maxCount) {
+        throwOrStoreError(`In ${indexName}.${collectionName}.${fieldName}, "minCount" can not be greater than "maxCount".`, verboseErrors, errors);
+      }
     }
 
-    if (fieldSpec.multivalued.hasOwnProperty('minCount') && fieldSpec.multivalued.hasOwnProperty('maxCount') && fieldSpec.multivalued.minCount > fieldSpec.multivalued.maxCount) {
-      throwOrStoreError(`In ${indexName}.${collectionName}.${fieldName}, "minCount" can not be greater than "maxCount".`, verboseErrors, errors);
+    if (errors.length > 0) {
+      return {isValid: false, errors: errors};
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Use shorthand for property "errors". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS3498)